test/position-managers/PositionManager.t.sol

```diff
@@ -0,0 +1,263 @@
+        uint256 balance0BeforeBurn = currency0.balanceOfSelf();
+        uint256 balance1BeforeBurn = currency1.balanceOfSelf();
+
+        BalanceDelta deltaDecrease = decreaseLiquidity(tokenId, liquidity, ZERO_BYTES);
+        burn(tokenId);
+
+        (liquidity,,) = manager.getPositionInfo(range.poolKey.toId(), positionId);
+
+        assertEq(liquidity, 0);
+
+        // TODO: slightly off by 1 bip (0.0001%)
+        assertApproxEqRel(
+            currency0.balanceOfSelf(), balance0BeforeBurn + uint256(uint128(deltaDecrease.amount0())), 0.0001e18
+        );
+        assertApproxEqRel(
+            currency1.balanceOfSelf(), balance1BeforeBurn + uint256(uint128(deltaDecrease.amount1())), 0.0001e18
+        );
+
+        // OZ 721 will revert if the token does not exist
+        vm.expectRevert();
+        lpm.ownerOf(1);
+
+        // no tokens were lost, TODO: fuzzer showing off by 1 sometimes
+        assertApproxEqAbs(currency0.balanceOfSelf(), balance0Start, 1 wei);
+        assertApproxEqAbs(currency1.balanceOfSelf(), balance1Start, 1 wei);
+    }
+
+    function test_decreaseLiquidity(IPoolManager.ModifyLiquidityParams memory params, uint256 decreaseLiquidityDelta)
+        public
+    {
+        uint256 tokenId;
+        (tokenId, params) = addFuzzyLiquidity(lpm, address(this), key, params, SQRT_PRICE_1_1, ZERO_BYTES);
+        vm.assume(0 < decreaseLiquidityDelta);
+        vm.assume(decreaseLiquidityDelta < uint256(type(int256).max));
+        vm.assume(int256(decreaseLiquidityDelta) <= params.liquidityDelta);
+
+        LiquidityRange memory range =
+            LiquidityRange({poolKey: key, tickLower: params.tickLower, tickUpper: params.tickUpper});
+
+        decreaseLiquidity(tokenId, decreaseLiquidityDelta, ZERO_BYTES);
+
+        bytes32 positionId =
+            keccak256(abi.encodePacked(address(lpm), range.tickLower, range.tickUpper, bytes32(tokenId)));
+        (uint256 liquidity,,) = manager.getPositionInfo(range.poolKey.toId(), positionId);
+
+        assertEq(liquidity, uint256(params.liquidityDelta) - decreaseLiquidityDelta);
+    }
+
+    function test_decreaseLiquidity_assertCollectedBalance(
+        IPoolManager.ModifyLiquidityParams memory params,
+        uint256 decreaseLiquidityDelta
+    ) public {
+        uint256 tokenId;
+        (tokenId, params) = addFuzzyLiquidity(lpm, address(this), key, params, SQRT_PRICE_1_1, ZERO_BYTES);
+        vm.assume(params.tickLower < 0 && 0 < params.tickUpper); // require two-sided liquidity
+        vm.assume(0 < decreaseLiquidityDelta);
+        vm.assume(decreaseLiquidityDelta < uint256(type(int256).max));
+        vm.assume(int256(decreaseLiquidityDelta) <= params.liquidityDelta);
+
+        LiquidityRange memory range =
+            LiquidityRange({poolKey: key, tickLower: params.tickLower, tickUpper: params.tickUpper});
+
+        // swap to create fees
+        uint256 swapAmount = 0.01e18;
+        swap(key, false, int256(swapAmount), ZERO_BYTES);
```

Comment:
i cant see anywhere that this test actually asserts anything about the fee amount being collected?

Reply:
it doesnt - this test just asserts the collected balance.. we do have other tests though that check fees

Reply:
would you prefer this test check that? its a bit difficult and there will probably be error because we will need to translate the fuzzed decreaseLiquidityDelta into amounts and then check that those amounts plus the expected fees are the new balance.. but I could add if you'd prefer

Reply:
no maybe not - just the comment "swap to create fees" made me think it was going to check that it generated fees. But I guess its more implicit here. And then explicit checks in others 👍